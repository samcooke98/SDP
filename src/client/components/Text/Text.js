import React from "react";
import styles from "./web.css";

export default class Text extends React.Component {
    render() {
        return (
            <p> {this.props.text} </p>
        )

    }

    someFunc = () => {
        //'This' is automatically bound when writing functions like this
        return JSON.stringify(this.props);
    }
}

TextInput.defaultProps = { 
    type: "text" 
}

