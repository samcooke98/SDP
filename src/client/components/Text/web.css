/* @import url('https://fonts.googleapis.com/css?family=Raleway'); */
@import url('https://fonts.googleapis.com/css?family=Raleway:400,700');

h3 { 
    font-size: 30px;
    font-weight: bold;
    margin: 12px;
}


p { 
    font-size: 16px; 
    letter-spacing: 2px;
}