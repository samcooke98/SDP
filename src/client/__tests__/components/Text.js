import React from "react";
import renderer from "react-test-renderer";
// import ReactDOM from 'react-dom';

import Text from "../../components/Text/Text.js";


describe('React Component: Text', () => {
    it("matches a snapshot", () => {
        const tree = renderer.create(
            <Text />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})