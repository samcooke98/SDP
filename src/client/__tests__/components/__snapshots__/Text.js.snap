// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Component: Text matches a snapshot 1`] = `
<p
  style={
    Object {
      "fontFamily": "Raleway, sans-serif",
    }
  }
>
   
   
</p>
`;
